proxy/src/main.rs

```diff
@@ -118,11 +119,34 @@ async fn main() -> anyhow::Result<()> {
     let mgmt_address: SocketAddr = arg_matches.value_of("mgmt").unwrap().parse()?;
     let http_address: SocketAddr = arg_matches.value_of("http").unwrap().parse()?;
 
+    // determine common name from tls-cert (-c server.crt param).
+    // used in asserting project name formatting invariant.
+    let common_name = match arg_matches.value_of("tls-cert") {
+        Some(cert_path) => {
+            let file = std::fs::File::open(cert_path).unwrap();
+            let almost_common_name = x509_parser::pem::Pem::read(std::io::BufReader::new(file))
+                .unwrap()
```

Comment:
`unwrap()` would cause a crash when triggered. It is better to use `?` so that an error message will be printed instead.

Reply:
how about in the next section, there is also an unwrap there as well.
Should I do this for both the unwraps above? (since the second one should always pass as I understand).

Reply:
>  Should I do this for both the unwraps above? (since the second one should always pass as I understand)

Yes, better fix both. It is easy to trigger second one as well -- try adding test on that case :)